dtorrent/src/tracker/http/http_handler.rs

```diff
@@ -61,28 +63,23 @@ impl HttpHandler {
     /// - There was a problem connecting to the tracker_url.
     /// - There was a problem writing to the tracker stream.
     /// - There was a problem reading the tracker stream.
-    pub fn https_request(&self) -> Result<Vec<u8>, HttpHandlerError> {
-        let connector = match TlsConnector::new() {
+    /// MAX ASYNC WITH TOKIO?
+    pub async fn https_request(&self) -> Result<Vec<u8>, HttpHandlerError> {
+        let from = match native_tls::TlsConnector::new() {
             Ok(connector) => connector,
             Err(err) => return Err(HttpHandlerError::CreateTlsConnectorError(err)),
         };
-        let stream = self.connect_tcp_stream()?;
-        let mut stream = match connector.connect(self.tracker_url.host.as_str(), stream) {
+        let connector = TlsConnector::from(from);
+        let stream = self.connect_tcp_stream().await?;
+
+        //TODO mejorar manejo de errores
```

Comment:
Yeah we should add thiserror for error handling. There is a ticker for that.

Reply:
This should be in English for consistency.